src/main/java/bit/javaoop/FixedBudgetManager.java

```diff
@@ -0,0 +1,37 @@
+package bit.javaoop;
+
+import java.util.ArrayList;
+
+public class FixedBudgetManager extends AbstractManager {
+    private double budget;
+
+    public FixedBudgetManager(String name, double salary, double budget) {
+        super(name, salary);
+        this.budget = budget;
+        this.employees = new ArrayList<>();
+    }
+
+    public double getBudget() {
+        return budget;
+    }
+
+    public void setBudget(double budget) {
+        this.budget = budget;
+    }
+
+    @Override
+    public boolean canHire(AbstractEmployee employee) {
+        return employee.getSalary() < getBudget();
+    }
+
+    @Override
+    public void hire(AbstractEmployee employee) {
+        this.employees.add(employee);
+        setBudget(getBudget() - employee.getSalary());
```

Comment:
Według mnie powinno to być tak:
`this.budget -= employee.getSalary();`
Używanie metod publicznych, które korzystają z pola prywatnego, do którego i tak masz dostęp i nie robią nic więcej poza ustawianiem oraz pobieraniem wartości jest nieefektywne i "brzydkie".